functests/1_performance/performance.go

```diff
@@ -340,8 +342,134 @@ var _ = Describe("[rfe_id:27368][performance]", func() {
 			Expect(err).ToNot(HaveOccurred(), fmt.Sprintf("cannot find MachineConfig object %s", initialKey.Name))
 		})
 	})
+
+	Context("Verify API Conversions", func() {
+		It("[test_id:35887] Verifies v1 <-> v1alpha1 conversions", func() {
+
+			By("Checking v1 -> v1alpha1 conversion")
+			v1alpha1Profile := &performancev1alpha1.PerformanceProfile{}
+			key := types.NamespacedName{
+				Name:      profile.Name,
+				Namespace: profile.Namespace,
+			}
+
+			err := testclient.GetWithRetry(context.TODO(), key, v1alpha1Profile)
+			ExpectWithOffset(1, err).ToNot(HaveOccurred(), "Failed getting v1alpha1Profile")
+			Expect(verifyV1alpha1Conversion(v1alpha1Profile, profile)).ToNot(HaveOccurred())
+
+			By("Checking v1alpha1 -> v1 conversion")
+			v1alpha1Profile.Name = "v1alpha"
+			v1alpha1Profile.ResourceVersion = ""
+			Expect(testclient.Client.Create(context.TODO(), v1alpha1Profile)).ToNot(HaveOccurred())
+
+			v1Profile := &performancev1.PerformanceProfile{}
+			key.Name = v1alpha1Profile.Name
+			err = testclient.GetWithRetry(context.TODO(), key, v1Profile)
+			ExpectWithOffset(1, err).ToNot(HaveOccurred(), "Failed getting v1profile")
+			Expect(verifyV1alpha1Conversion(v1alpha1Profile, v1Profile)).ToNot(HaveOccurred())
+
+			Expect(testclient.Client.Delete(context.TODO(), v1Profile)).ToNot(HaveOccurred())
+		})
+	})
 })
 
+func verifyV1alpha1Conversion(v1alpha1Profile *performancev1alpha1.PerformanceProfile, v1Profile *performancev1.PerformanceProfile) error {
+	specCPU := v1alpha1Profile.Spec.CPU
+	if (specCPU == nil) != (v1Profile.Spec.CPU == nil) {
```

Comment:
here and below: this looks a bit convoluted: why not

  if (specCPU == nil) && (v1Profile.Spec.CPU != nil) {

?

Reply:
You will need both ways and it will also lead on more verbose code  below instead of simple "if specCPU != nil".
The (specCPU == nil) != (v1Profile.Spec.CPU == nil) micro-pattern checks one of them has a value and the other one has none.

Bellow we can then have 2 cases: both are nil => equal or both have value => we can safely start check them

Reply:
ok, makes sense